demo/admin/src/common/MasterMenu.tsx

```diff
@@ -1,69 +1,169 @@
+    },
+];
 
-export const MasterMenu: React.FC = () => {
-    const { open, toggleOpen } = React.useContext(MenuContext);
-    const windowSize = useWindowSize();
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export const pageTreeDocumentTypes: Record<string, DocumentInterface<any, any>> = {
+    Page,
+    Link,
+};
+const RedirectsPage = createRedirectsPage({ scopeParts: ["domain"] });
+
+const getMasterMenuData = ({ brevoContactConfig }: { brevoContactConfig: BrevoContactConfig }): MasterMenuData => {
+    const BrevoContactsPage = createBrevoContactsPage({
+        additionalAttributesFragment: brevoContactConfig.additionalAttributesFragment,
+        additionalGridFields: brevoContactConfig.additionalGridFields,
+        additionalFormFields: brevoContactConfig.additionalFormFields,
+        input2State: brevoContactConfig.input2State,
+    });
+
+    const TargetGroupsPage = createTargetGroupsPage({
+        additionalFormFields: additionalFormConfig.additionalFormFields,
+        exportTargetGroupOptions: {
+            additionalAttributesFragment: brevoContactConfig.additionalAttributesFragment,
+            exportFields: brevoContactConfig.exportFields,
+        },
+        nodeFragment: additionalFormConfig.nodeFragment,
+        input2State: additionalFormConfig.input2State,
+    });
+
+    return [
+        {
+            type: "route",
+            primary: <FormattedMessage id="menu.dashboard" defaultMessage="Dashboard" />,
+            icon: <Dashboard />,
+            route: {
+                path: "/dashboard",
+                component: DashboardPage,
+            },
+        },
+        {
+            type: "route",
+            primary: <FormattedMessage id="menu.pageTree" defaultMessage="Page tree" />,
+            icon: <PageTree />,
+            route: {
+                path: "/pages/pagetree/main-navigation",
+                render: () => (
+                    <PagesPage
+                        path="/pages/pagetree/main-navigation"
+                        allCategories={pageTreeCategories}
+                        documentTypes={pageTreeDocumentTypes}
+                        category="MainNavigation"
+                        renderContentScopeIndicator={(scope) => <ContentScopeIndicator scope={scope} />}
+                    />
+                ),
+            },
+            requiredPermission: "pageTree",
+        },
+        {
+            type: "collapsible",
+            primary: <FormattedMessage id="menu.newsletter" defaultMessage="Newsletter" />,
+            icon: <Mail />,
+            items: [
+                {
+                    type: "route",
+                    primary: <FormattedMessage id="menu.newsletter.emailCampaigns" defaultMessage="Email campaigns" />,
+                    route: {
+                        path: "/newsletter/email-campaigns",
+                        component: () => <EmailCampaignsPage EmailCampaignContentBlock={EmailCampaignContentBlock} />,
```

Comment:
```suggestion
                        render: () => <EmailCampaignsPage EmailCampaignContentBlock={EmailCampaignContentBlock} />,
```

This must be render or the component will unmount on every render: https://v5.reactrouter.com/web/api/Route/component#:~:text=That%20means%20if%20you%20provide%20an%20inline%20function%20to%20the%20component%20prop%2C%20you%20would%20create%20a%20new%20component%20every%20render.